libs/trainer/mlpe/trainer/trainer.py

```diff
@@ -110,15 +110,14 @@ def train(
     max_epochs: int = 40,
     init_weights: Optional[Path] = None,
     lr: float = 1e-3,
-    early_stop: int = 20,
+    min_lr: float = 1e-5,
+    decay_steps: int = 10000,
+    weight_decay: float = 0.0,
+    early_stop: Optional[int] = None,
     # misc params
     device: Optional[str] = None,
     use_amp: bool = False,
     profile: bool = False,
-    optimizer_fn: Callable = torch.optim.Adam,
-    optimizer_kwargs: dict = dict(weight_decay=0),
-    scheduler_fn: Callable = torch.optim.lr_scheduler.CosineAnnealingLR,
-    scheduler_kwargs: dict = dict(eta_min=1e-5, T_max=10000),
```

Comment:
Why take this out? Let's just remove the lines from the project config now until the typeo trick is discovered.

Reply:
Would unfortunately require additional `typeo` development as far as I'm aware. For now let's stick with the hard coded values(feel free to change it to exponential, or whatever you want).

Reply:
It does not break anything right now.

Reply:
You just need to not mention it in the config

Reply:
Only because `typeo` is using the specified default. If you were to pass it as a config in the `pyproject.toml,` it would break. Since using the default is essentially equivalent to hard coding, lets just do that for now

Reply:
I guess my philosophy is that the everything in the production repo should work as expected. As a hack in a branch this is of course totally fine.

Reply:
your philosophy is respected even without this change, and requires no breaking change in terms of typeo.